src/boot/efi/util.c

```diff
@@ -760,12 +760,12 @@ EFI_STATUS make_file_device_path(EFI_HANDLE device, const char16_t *file, EFI_DE
         /* Point dp to the end node of the copied device path. */
         dp = (EFI_DEVICE_PATH *) ((uint8_t *) *ret_dp + dp_size - END_DEVICE_PATH_LENGTH);
 
-        /* Replace end node with file media device path. */
-        FILEPATH_DEVICE_PATH *file_dp = (FILEPATH_DEVICE_PATH *) dp;
-        file_dp->Header.Type = MEDIA_DEVICE_PATH;
-        file_dp->Header.SubType = MEDIA_FILEPATH_DP;
-        memcpy(&file_dp->PathName, file, file_size);
-        SetDevicePathNodeLength(&file_dp->Header, SIZE_OF_FILEPATH_DEVICE_PATH + file_size);
+        /* Replace end node with file media device path. Use memcpy in case dp is unaligned (if accessed as
+         * FILEPATH_DEVICE_PATH). */
+        dp->Type = MEDIA_DEVICE_PATH;
+        dp->SubType = MEDIA_FILEPATH_DP;
+        memcpy((uint8_t *) dp + offsetof(FILEPATH_DEVICE_PATH, PathName), file, file_size);
+        SetDevicePathNodeLength(dp, SIZE_OF_FILEPATH_DEVICE_PATH + file_size);
```

Comment:
looking in the headers, `SIZE_OF_FILEPATH_DEVICE_PATH` is defined as `EFI_FIELD_OFFSET(FILEPATH_DEVICE_PATH, PathName)`, hence equivalent to `offsetof(FILEPATH_DEVICE_PATH, PathName)`. The `memcpy()` call uses the latter, here you use the former. I'd stick to one way to write this. Probably use the offsetof() thing, simply because it's more readable.